### ISSUE TITLE
Impossible to edit a Job template of you leave the inventory field empty

### ISSUE THREAD

**Issue by astraios** [ansible/awx#415]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - UI

##### SUMMARY
A job template is impossible to edit if you leave the inventory field empty and check the "prompt on launch" option.

##### ENVIRONMENT

* AWX version: AWX 1.0.1.10
* AWX install method: docker on linux with the install playbook
* Ansible version:  2.4
* Operating System: Centos7
* Web Browser: Chrome Version 61.0.3163.100

##### STEPS TO REPRODUCE

* Create a new Job template. Fill the fields except for inventory and check "prompt on launch" for this one.
* Save, nothing happens even if the job should appear to the template list, save again and an error pop out saying "JobTemplate with this (polymorphic_ctype, name) combination already exists.".
* Refresh the page, or go the Template section through the menu.
* Click on the Job and nothing happens. 

##### EXPECTED RESULTS

* First, when saving the new job, it should appear in the template list
* Then you should be able to edit the job

##### ACTUAL RESULTS

* You can see the job in the template list but can do nothing about it except deleting it.

**Comment by matburt** [comment]
Can you show us what the job template looks like from the api (visit `towerhost/api/v2/job_templates/<id>/`

Also... please provide the js console error that happened when saving the job template

**Comment by astraios** [comment]
Hi Mat,

Here, the result from the API

```
HTTP 200 OK
Allow: GET, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept
X-API-Node: awx
X-API-Time: 0.323s




{
    "id": 42,
    "type": "job_template",
    "url": "/api/v2/job_templates/42/",
    "related": {
        "named_url": "/api/v2/job_templates/github issue/",
        "created_by": "/api/v2/users/2/",
        "modified_by": "/api/v2/users/2/",
        "labels": "/api/v2/job_templates/42/labels/",
        "project": "/api/v2/projects/15/",
        "extra_credentials": "/api/v2/job_templates/42/extra_credentials/",
        "notification_templates_error": "/api/v2/job_templates/42/notification_templates_error/",
        "notification_templates_success": "/api/v2/job_templates/42/notification_templates_success/",
        "jobs": "/api/v2/job_templates/42/jobs/",
        "object_roles": "/api/v2/job_templates/42/object_roles/",
        "notification_templates_any": "/api/v2/job_templates/42/notification_templates_any/",
        "access_list": "/api/v2/job_templates/42/access_list/",
        "launch": "/api/v2/job_templates/42/launch/",
        "instance_groups": "/api/v2/job_templates/42/instance_groups/",
        "schedules": "/api/v2/job_templates/42/schedules/",
        "activity_stream": "/api/v2/job_templates/42/activity_stream/",
        "survey_spec": "/api/v2/job_templates/42/survey_spec/"
    },
    "summary_fields": {
        "project": {
            "id": 15,
            "name": "base - global setup",
            "description": "",
            "status": "successful",
            "scm_type": "git"
        },
        "created_by": {
            "id": 2,
            "username": "admin",
            "first_name": "admin",
            "last_name": "istrator"
        },
        "modified_by": {
            "id": 2,
            "username": "admin",
            "first_name": "admin",
            "last_name": "istrator"
        },
        "object_roles": {
            "admin_role": {
                "id": 196,
                "description": "Can manage all aspects of the job template",
                "name": "Admin"
            },
            "execute_role": {
                "id": 195,
                "description": "May run the job template",
                "name": "Execute"
            },
            "read_role": {
                "id": 194,
                "description": "May view settings for the job template",
                "name": "Read"
            }
        },
        "user_capabilities": {
            "edit": true,
            "start": true,
            "copy": true,
            "schedule": true,
            "delete": true
        },
        "labels": {
            "count": 0,
            "results": []
        },
        "recent_jobs": [],
        "extra_credentials": []
    },
    "created": "2017-10-12T17:19:15.531Z",
    "modified": "2017-10-12T17:19:15.531Z",
    "name": "github issue",
    "description": "",
    "job_type": "run",
    "inventory": null,
    "project": 15,
    "playbook": "ssh.yml",
    "credential": null,
    "vault_credential": null,
    "forks": 0,
    "limit": "",
    "verbosity": 0,
    "extra_vars": "",
    "job_tags": "",
    "force_handlers": false,
    "skip_tags": "",
    "start_at_task": "",
    "timeout": 0,
    "use_fact_cache": false,
    "last_job_run": null,
    "last_job_failed": false,
    "next_job_run": null,
    "status": "never updated",
    "host_config_key": "",
    "ask_diff_mode_on_launch": false,
    "ask_variables_on_launch": false,
    "ask_limit_on_launch": false,
    "ask_tags_on_launch": false,
    "ask_skip_tags_on_launch": false,
    "ask_job_type_on_launch": false,
    "ask_verbosity_on_launch": false,
    "ask_inventory_on_launch": true,
    "ask_credential_on_launch": true,
    "survey_enabled": false,
    "become_enabled": false,
    "diff_mode": false,
    "allow_simultaneous": false
}
```

The js console output, I hope it's enough: 
<img width="699" alt="capture d ecran 2017-10-12 a 19 19 40" src="https://user-images.githubusercontent.com/16399198/31509794-00497f84-af83-11e7-80bd-ce1579d8d05c.png">


Regards,
Thomas

**Comment by matburt** [comment]
I just forced an image rebuild of 1.0.1.10 and pushed it to dockerhub... can you re-pull the image and try it out?

**Comment by astraios** [comment]
I did and got the same problem and same js output and also the same behavior.
Thx

**Comment by matburt** [comment]
We've fixed the images, 1.0.1.12+ should work good for you.
